src/main/java/com/online/cinema/entity/moviesession/dao/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.online.cinema.entity.moviesession.dao;
+
+import com.online.cinema.dao.GenericDaoImpl;
+import com.online.cinema.entity.moviesession.model.MovieSession;
+import com.online.cinema.exceptions.DataProcessingException;
+import com.online.cinema.lib.Dao;
+import com.online.cinema.util.HibernateUtil;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+
+@Dao
+public class MovieSessionDaoImpl extends GenericDaoImpl<MovieSession>
+        implements MovieSessionDao {
+
+    @Override
+    public List<MovieSession> getAll() {
+        return super.getAll(MovieSession.class);
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<MovieSession> query
+                    = criteriaBuilder.createQuery(MovieSession.class);
+            Root<MovieSession> root = query.from(MovieSession.class);
+            Predicate idPredicate
+                    = criteriaBuilder.equal(root.get("movie"), movieId);
+            Predicate datePredicate = criteriaBuilder.between(
+                    root.get("showTime"), date.atStartOfDay(), date.atTime(LocalTime.MAX));
+            return session.createQuery(query.where(idPredicate, datePredicate)).getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException(
+                    "There was an error retrieving available sessions", e);
```

Comment:
let's make exception messages more informative (add info about input parameters that you've used when error occurred)